### ISSUE TITLE
Configuring Two-Factor

### ISSUE THREAD

**Issue by ntallman** [sovereign/sovereign#673]
The ansible playbook enables using Google Authenticator for two-factor authentication when connecting via ssh. However there are no instructions for configuring Google Authenticator and I cannot ssh to my server now because I don't have the  verification code. Am I missing something? Thanks.

**Comment by tomw1808** [comment]
The keys are created and put in ~/.google_authenticator. When you run the playbook then there is one time a 5 seconds timeout which _should_ show you that the keys were copied to the tmp folder (/tmp/sovereign-google-auth-files). Unfortuantely the promt from file roles/common/tasks/google_auth.yml is never shown - at least not for me. I was luckily still logged into the instance where I ran the playbook to, so I had access to the keys there.

Maybe that helps.

**Comment by adborden** [comment]
Yeah, the secrets are copied locally to `/tmp/sovereign-google-auth-files`, the machine from which you are running ansible. If you missed it, you can re-run the playbook and the keys will be copied locally again.

**Comment by TinkerFu** [comment]
Same issue.  Found the keys yet still cannot log in.  Tried creating a new key, same issue.   Is there an easy way to disable this?

**Comment by tirth78** [comment]
I would like to work on this issue. I am a Masters student in BITS Pilani. It would be really helpful if I could get some kind of approval from the owner/author.

**Comment by neuffer** [comment]
> I would like to work on this issue. I am a Masters student in BITS Pilani. It would be really helpful if I could get some kind of approval from the owner/author.

If you work on this, maybe you could also look at integration 2Factor authentication into a few more pieces of the software stack, for example postfixadmin?

**Comment by tirth78** [comment]
Could you please give some resources about the same?

**Comment by neuffer** [comment]
> Could you please give some resources about the same?
> […](#)

You could take a look at the Postfixadmin repository, somebody started working at this a while ago.
[https://github.com/postfixadmin/postfixadmin/pull/322](https://github.com/postfixadmin/postfixadmin/pull/322)

Get it into the upstream Postfixadmin and it will be available in Sovereign.
